packages/ERTP/src/issuer.chainmail

```diff
@@ -0,0 +1,209 @@
+ */
+interface Purse (Amount) {
+
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+
+  /** Get the amount contained in this purse, confirmed by the issuer. */
+  getBalance() -> (Amount);
+
+  /**
+   * Deposit all the contents of srcPayment into this purse, returning the
+   * amount. If the optional argument amount does not equal the balance of srcPayment,
+   * throw error.
+   */
+  deposit(srcPayment :Payment, amount :Amount) -> (Amount);
+
+  /** Withdraw amount from this purse into a new Payment. */
+  withdraw(amount :Amount) -> (Payment);
+
+}
+
+/**
+ * Payments hold amount of digital assets of the same brand in transit. Payments can
+ * be deposited in purses, split into multiple payments, combined, and
+ * claimed (getting an exclusive payment). Payments are linear, meaning
+ * that either a payment has its full original balance, or it is used up
+ * entirely. It is impossible to partially use a payment. 
+ *
+ * Payments are often received from other actors and therefore should
+ * not be trusted themselves. To get the balance of a payment, use the
+ * trusted issuer: issuer.getBalance(payment),
+ *
+ * Payments can be converted to Purses by getting a trusted issuer and
+ * calling `issuer.makeEmptyPurse()` to create a purse, then `purse.deposit(payment)`.
+ */
+interface Payment (Amount) {
+
+  /** 
+   * Get the allegedBrand, indicating the kind of digital asset this
+   * payment purports to be, and which issuer to use. Because payments 
+   * are not trusted, any method calls on payments should be treated
+   * with suspicion and verified elsewhere.
+   */ 
+  allegedBrand() -> (Brand);
+}
+
+/** 
+ * All of the difference in how digital asset amount are manipulated can be reduced to
+ * the behavior of the math on extents. We extract this
+ * custom logic into mathHelpers. MathHelpers are about extent
+ * arithmetic, whereas AmountMath is about amounts, which are the
+ * extents labeled with a brand. AmountMath use mathHelpers to do their extent arithmetic,
+ * and then brand the results, making a new amount. 
+ */ 
+interface mathHelpers () {
+  /** 
+   * Check the kind of this extent and throw if it is not the
+   * expected kind. 
+   */
+  doAssertKind(allegedExtent :Extent) -> (Extent);
+
+  /**
+   * Get the representation for the identity element (often 0 or an
+   * empty array)
+   */
+  doGetIdentity() -> (Extent);
+
+  /** Is the extent the identity element? */
+  doIsIdentity(extent :Extent) -> (boolean);
+
+  /** Is the left greater than or equal to the right? */
+  doIsGTE(left :Extent, right :Extent) -> (boolean);
+
+  /** Does left equal right? */
+  doIsEqual(left :Extent, right :Extent) -> (boolean);
+
+  /** Return the left combined with the right */
+  doAdd(left :Extent, right :Extent) -> (Extent);
+
+  /**
+   * Return what remains after removing the right from the left. If
+   * the result is negative (i.e. something in the right was not in the
```

Comment:
Conventional set subtraction has no negatives, so this explanation doesn't explain how this differs. Use the wording we agreed on somewhere else.